frontend/src/views/AssignmentOverview.tsx

```diff
@@ -28,65 +28,73 @@ interface AssignmentOverviewProps {
 
 export default function AssignmentOverview(props: AssignmentOverviewProps) {
   const classes = useStyles();
-  const [ assignments, setAssignments ] = useState([])
-  const [ deployedAssignment, setDeployedAssignment ] = useState("")
-  const [load, setLoad] = useState(true)  
+  const [assignments, setAssignments] = useState([])
+  const [deployedAssignment, setDeployedAssignment] = useState("")
+  const [load, setLoad] = useState(true)
   const [deploymentNotificationOpen, setDeploymentNotificationOpen] = React.useState(false);
 
   useEffect(() => {
     setLoad(false)
-    fetch(APIRequest("/api/user/assignments", {   headers:{  authorization: localStorage.getItem("token") || ""}}))
+    fetch(APIRequest("/api/user/assignments", { headers: { authorization: localStorage.getItem("token") || "" } }))
       .then(res => res.json())
       .then(setAssignments)
-  },[load])
+  }, [load])
 
   const createEnvironment = useCallback(async (assignment: string) => {
-    const result = await fetch(APIRequest(`/api/environment/create?environment=${assignment}`, {
-      method: 'POST', 
-      headers: {'Content-Type': 'application/json', authorization: localStorage.getItem("token") || ""} 
-    }))
-    if (result.status === 200){
-      setDeploymentResult("Deployment successful!")
-      setDeploymentSeverity("success")
-      setDeploymentNotificationOpen(true)
-      setDeployedAssignment(assignment)
-    } else {
-      const a = await result.json()
-      setDeploymentResult("Deployment failed! (" + a.message + ")")
+    setDeploymentResult("Starting deployment...")
+    setDeploymentSeverity("info")
+    setDeploymentNotificationOpen(true)
+    try {
+      const result = await fetch(APIRequest(`/api/environment/create?environment=${assignment}`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json', authorization: localStorage.getItem("token") || "" }
+      }))
+      if (result.status === 200) {
+        setDeploymentResult("Deployment successful!")
```

Comment:
You can most likely use one call and initialize state to a default object, and just pass the new object in that one call.

Reply:
tried that. If it looks good, maybe this can ge applied to other notifications also